fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingByRecords.java

```diff
@@ -624,7 +625,10 @@ private TupleRange computeRecordsRange() {
             RecordCursor<Tuple> cursor = common.getRecordStoreBuilder().copyBuilder().setContext(context).open()
                     .getPrimaryKeyBoundaries(tupleRange.getLow(), tupleRange.getHigh());
             return context.asyncToSync(FDBStoreTimer.Waits.WAIT_GET_BOUNDARY, cursor.asList());
-        });
+        }, Lists.newArrayList(
+                LogMessageKeys.TRANSACTION_NAME, "IndexingByRecords::getPrimaryKeyBoundaries",
```

Comment:
Looking at how essentially all of these runner creation thingies want to log the transaction name, there's another part of me that wonders if the right approach is to add `name` to the `FDBRecordContextConfig` and then log that in `FDBDatabaseRunnerImpl` (and probably expose it in the `FDBDatabaseRunner` somehow). We'd need to make sure to qualify what makes it different from a transaction's ID (i.e., descriptive name to help the user differentiate different sources of transactions vs unique ID to allow different transaction's to be distinguished in the logs), but it could be more generally useful.

That being said, I think that adding the general purpose "additional key values" is probably also good, so I think we could take this PR and then add names to `FDBRecordContextConfig`s (later).

Reply:
Yeah, I was thinking something very similar to this.  The main argument against that approach might be that the runner can be created once, then used many times, and each use might logically want a different name.  Either way, I'm not really pleased with the solution as it is now.